openapi/src/main/java/io/micronaut/openapi/visitor/ElementUtils.java

```diff
@@ -0,0 +1,144 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.openapi.visitor;
+
+import java.io.File;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.Future;
+
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.inject.ast.ClassElement;
+import io.micronaut.inject.ast.TypedElement;
+
+/**
+ * Some util methods.
+ *
+ * @since 4.8.3
+ */
+public final class ElementUtils {
+
+    public static final List<String> CONTAINER_TYPES = Arrays.asList(
+        Optional.class.getName(),
+        Future.class.getName(),
+        "org.reactivestreams.Publisher",
+        "io.reactivex.Single",
+        "io.reactivex.Observable",
+        "io.reactivex.Maybe",
+        "io.reactivex.rxjava3.core.Single",
+        "io.reactivex.rxjava3.core.Observable",
+        "io.reactivex.rxjava3.core.Maybe",
+        "kotlinx.coroutines.flow.Flow"
+    );
+
+    public static final List<String> FILE_TYPES = Arrays.asList(
+        // this class from micronaut-http-server
+        "io.micronaut.http.server.types.files.FileCustomizableResponseType",
+        File.class.getName(),
+        InputStream.class.getName(),
+        ByteBuffer.class.getName()
+    );
+
+    public static final List<String> VOID_TYPES = Arrays.asList(
+        void.class.getName(),
+        Void.class.getName(),
+        "kotlin.Unit"
+    );
+
+    private ElementUtils() {
+    }
+
+    /**
+     * Checks Nullable annotions / optinal type to understand that the element can be null.
+     *
+     * @param element typed element
+     *
+     * @return true if element is nullable, false - otherwise.
+     */
+    public static boolean isNullable(TypedElement element) {
+        return element.isNullable()
+            || element.getType().isOptional()
+            || element.hasStereotype(Nullable.class)
+            || element.hasStereotype("javax.annotation.Nullable")
+            || element.hasStereotype("jakarta.annotation.Nullable")
+            || element.hasStereotype("org.jetbrains.annotations.Nullable")
+            || element.hasStereotype("androidx.annotation.Nullable")
+            || element.hasStereotype("edu.umd.cs.findbugs.annotations.Nullable")
+            || element.hasStereotype("org.eclipse.jdt.annotation.Nullable")
+            || element.hasStereotype("io.reactivex.annotations.Nullable")
+            || element.hasStereotype("io.reactivex.rxjava3.annotations.Nullable")
+            || element.hasStereotype("reactor.util.annotation.Nullable")
+            || element.hasStereotype("org.jspecify.annotations.Nullable");
```

Comment:
hard coding this is not great, these should be replaced by classes like this:

https://github.com/micronaut-projects/micronaut-core/blob/4.0.x/core-processor/src/main/java/io/micronaut/inject/annotation/internal/JakartaNullableTransformer.java

Perhaps in core

Reply:
@graemerocher By the way, how is this solution worse than what you suggest? You propose to hardcode the names of annotations separated by classes, I have them in one place, without using additional complex structures and the check goes linearly. In both cases, you will have to hardcode the full path to the annotation class, only in your case I will have to describe 10 more classes without any profit.

In general, I do not understand the better your solution :-)

Reply:
because if it is part of core then every module does not have to repeat the same logic and it is encapsulated into a single place. Other parts of the framework will likely need to know what is nullable and that could differ from what OpenAPI believes is nullable if it is only captured here.

Reply:
@graemerocher do you suggest to add these annotaions to core module?

Reply:
that is what makes sense indeed